tests/cmdline/params/types/test_code.py

```diff
@@ -7,109 +7,127 @@
-        Appending the special ambiguity breaker character will force the identifier to be treated as a LABEL
-        """
-        identifier = '{}'.format(self.entity_02.label)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-        identifier = '{}{}'.format(self.entity_02.label, OrmEntityLoader.label_ambiguity_breaker)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_02.uuid)
-
-    def test_ambiguous_label_uuid(self):
-        """
-        Situation: LABEL of entity_03 is exactly equal to UUID of entity_01
-
-        Verify that using an ambiguous identifier gives precedence to the UUID interpretation
-        Appending the special ambiguity breaker character will force the identifier to be treated as a LABEL
-        """
-        identifier = '{}'.format(self.entity_03.label)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_01.uuid)
-
-        identifier = '{}{}'.format(self.entity_03.label, OrmEntityLoader.label_ambiguity_breaker)
-        result = self.param_base.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_03.uuid)
-
-    def test_entry_point_validation(self):
-        """Verify that when an `entry_point` is defined in the constructor, it is respected in the validation."""
-        identifier = '{}'.format(self.entity_02.pk)
-        result = self.param_entry_point.convert(identifier, None, None)
-        self.assertEqual(result.uuid, self.entity_02.uuid)
-
-        with self.assertRaises(click.BadParameter):
-            identifier = '{}'.format(self.entity_03.pk)
-            result = self.param_entry_point.convert(identifier, None, None)
+@pytest.fixture
+def setup_codes(clear_database_before_test, aiida_localhost):
+    """Create some `Code` instances to test the `CodeParamType` parameter type for the command line infrastructure.
+
+    We create an initial code with a random name and then on purpose create two code with a name that matches exactly
+    the ID and UUID, respectively, of the first one. This allows us to test the rules implemented to solve ambiguities
+    that arise when determing the identifier type.
+    """
+    entity_01 = Code(remote_computer_exec=(aiida_localhost, '/bin/true')).store()
+    entity_02 = Code(remote_computer_exec=(aiida_localhost, '/bin/true'), input_plugin_name='arithmetic.add').store()
+    entity_03 = Code(remote_computer_exec=(aiida_localhost, '/bin/true'), input_plugin_name='templatereplacer').store()
+
+    entity_01.label = 'computer_01'
+    entity_02.label = str(entity_01.pk)
+    entity_03.label = str(entity_01.uuid)
+
+    return entity_01, entity_02, entity_03
+
+
+def test_get_by_id(setup_codes):
+    """Verify that using the ID will retrieve the correct entity."""
+    entity_01, entity_02, entity_03 = setup_codes
```

Comment:
```suggestion
    entity_01, _, _ = setup_codes
```

Reply:
I decided to actually purposefully keep these lines like this, as to have them identical in each test. I think this makes it easier to see that this is just a common setup line done in each test. That's why I added the pylint disable statement `unused-variable` to the module. Would it be ok to keep these?

Reply:
Fair enough. I don't see that it is necessary to do it this way, but if you prefer it, I guess it's fine. The danger, of course, is that the disabled linting feature will mask some other unused variable. And if it was just to make it easier to copy-paste, I believe I have solved that with my diligent suggestions 😄 😅

Reply:
it certainly is not necessary, but find it just easier to see that all these lines are just the same. If they start having random `_` interspersed, that interpretation becomes more difficult. Given that this is a test module, having the global pylint disable statement is also no threat I feel. In any case, it certainly isn't because I am lazy and prefer simple copy pasting :)

Reply:
I don't feel the exact same way about this as you do, and would rather not have the blind pylint disable in exchange for the same line copied over and over - even if it is a test file, and maybe even more so, since it's usually in those that one most easily gets lost 😅
However, as this mostly comes down to personal taste, and doesn't affect the code in any significant way, I will note this down as a mark of the author on the code, which may then stand as a testamente or otherwise 😉